docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst

```diff
@@ -1811,66 +1810,86 @@ for valid upgrades, :ref:`here <upgrade-model-reference>`.
+from all packages with the name ``package-name`` known to the Ledger API.
 
-.. _dynamic-package-resolution-in-command-submission:
+`By-package-name template ID` addressing is supported for all upgradable packages (LF >= 1.17)
+and is possible on both the write path (command submission) and read path (Ledger API queries).
 
-Dynamic Package Resolution in Command Submission
-------------------------------------------------
+The `by-package-name template ID` is only a Ledger API interface concept.
+Internally, Canton continues to use the fully qualified template ID for all operations.
+Therefore, when the new format is used in requests to the Ledger API,
+a dynamic resolution is performed as described in the following sections.
 
-Dynamic package resolution can happen in two cases during command
-submission:
+.. _package-preference:
 
--  For command submissions that use the package-name selector
-   (``#<package-name>``) in the command’s templateId field (e.g. in a
-   create command :ref:`here <com.daml.ledger.api.v1.CreateCommand>`)
+Package Preference
+------------------
 
--  For command submissions leading to Daml transactions that contain
-   actions exercised on interfaces. In this situation there may be
-   many versions of a template that implement the interface being
-   exercised.
+On processing a command submission, there are scenarios where the Ledger API server needs
+to disambiguate `by-package-name template IDs` to `by-package-id template IDs` (see :ref:`dynamic-package-resolution-in-command-submission`).
+For this purpose, the **package preference** concept is introduced
+as the mapping from package-name to package ID for all known package-names on the participant.
 
-In these situations the following rules are followed to resolve the
-package-name to a package-id:
+The package preference is needed at each command submission time and is assembled from two sources:
 
--  By default, the participant resolves a package-name to the package-id
-   pertaining to the highest package version uploaded
+- The **default** package preference of the Ledger API to which the command is submitted.
+  This is the exhaustive mapping from package-name to package ID for all known package-names on the participant.
+  If multiple package IDs exist for a package-name, the package ID pertaining to the package with the highest
+  version is used.
 
--  The command submitter can override the above-mentioned default
-   participant resolution by pinning package-ids in the Command’s
-   :ref:`package_id_selection_preference <com.daml.ledger.api.v1.Commands.package_id_selection_preference>`.
-   More specifically, this field is a list of package-ids that must
-   be explicitly used when resolving package-name *ambiguities* in
-   either command template-id or interface resolution.
+- The Ledger API client **override** as the
+  :ref:`package_id_selection_preference <com.daml.ledger.api.v1.Commands.package_id_selection_preference>` in a command submission.
```

Comment:
This reads weird. Can you find a formulation that does not use "the override" as a noun?

Perhaps "the package-id selection preference list specified in the submitted command".